indico_toolkit/association/positioning.py

```diff
@@ -164,6 +165,29 @@ def get_vertical_overlap(self, pos1: dict, pos2: dict) -> float:
             return vertical_overlap_distance / position_height
         else:
             return 0.0
+    
+    def tokens_within_bounds(self, bbox: dict, ocr_tokens: List[dict], include_overlap: bool=False):
+        """
+        Accept a dict of bounding box dimensions with a page number,
+        along with a list of ocr tokens either from raw or OnDoc object.
+        Return all tokens that lie within bounding box.
+        Tokens with partial overlap are excluded by default.
+        """
+        if "position" not in ocr_tokens[0].keys() or "page_num" not in ocr_tokens[0].keys():
```

Comment:
this is good-- but fyi `.keys()` is not needed for checking if key in dict

Reply:
Noted, removed.